pageserver/src/repository.rs

```diff
@@ -81,7 +89,49 @@ pub trait Timeline {
         tablespace_id: Oid,
         src_db_id: Oid,
         src_tablespace_id: Oid,
-    ) -> Result<()>;
+    ) -> Result<()> {
+        let mut n = 0;
+        for forknum in &[
+            pg_constants::MAIN_FORKNUM,
+            pg_constants::FSM_FORKNUM,
+            pg_constants::VISIBILITYMAP_FORKNUM,
+            pg_constants::INIT_FORKNUM,
+            pg_constants::PG_FILENODEMAP_FORKNUM,
+        ] {
+            let key = RepositoryKey {
+                tag: BufferTag {
+                    rel: RelTag {
+                        spcnode: src_tablespace_id,
+                        dbnode: src_db_id,
+                        relnode: 0,
+                        forknum: *forknum,
+                    },
+                    blknum: 0,
+                },
+                lsn: Lsn(0),
+            };
+            let mut iter = self.iterator();
+            iter.first(&key);
+            while iter.valid() {
```

Comment:
This is very non-standard. Iterators should impl the `Iterator` trait, then this code can be written the standard way.